crates/red_knot_python_semantic/src/types.rs

```diff
@@ -3229,3 +3230,176 @@ mod tests {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod property_tests {
+    use crate::types::Type;
+
+    use super::tests::{setup_db, Ty};
+    use crate::Db;
+    use quickcheck::{Arbitrary, Gen};
+    use quickcheck_macros::quickcheck;
+
+    fn arbitrary_core_type(g: &mut Gen) -> Ty {
+        let int_lit = Ty::IntLiteral(i64::arbitrary(g));
+        let bool_lit = Ty::BooleanLiteral(bool::arbitrary(g));
+        g.choose(&[
+            Ty::Never,
+            Ty::Unknown,
+            Ty::None,
+            Ty::Any,
+            Ty::Todo,
+            int_lit,
+            bool_lit,
+            Ty::StringLiteral(""),
+            Ty::StringLiteral("a"),
+            Ty::LiteralString,
+            Ty::BytesLiteral(""),
+            Ty::BytesLiteral("\x00"),
+            Ty::BuiltinInstance("str"),
+            Ty::BuiltinInstance("int"),
+            Ty::BuiltinInstance("bool"),
+            Ty::BuiltinInstance("object"),
+            Ty::BuiltinClassLiteral("str"),
+            Ty::BuiltinClassLiteral("int"),
+            Ty::BuiltinClassLiteral("bool"),
+            Ty::BuiltinClassLiteral("object"),
+        ])
+        .unwrap()
+        .clone()
+    }
+
+    fn arbitrary_type(g: &mut Gen, size: u32) -> Ty {
+        if size == 0 {
+            arbitrary_core_type(g)
+        } else {
+            match u32::arbitrary(g) % 4 {
```

Comment:
There must be a better way to do this. If only we had the author of this library on our team...

Reply:
CC: @BurntSushi